### PR TITLE
Implement 1.0.7 of libddwaf

### PR DESCRIPTION
**Pull request by robertpi**
Our C++ WAF implementation, libddwaf - previously PowerWaf / libSqreen,
has been though a big clean up of their APIs and the format of
configuration file. This commit synchorinizes the API signatures and
naming of methods / objects.

The details of the new API can be found here:
https://github.com/DataDog/libddwaf/blob/master/include/ddwaf.h

### CODE REVIEW

File: test/Datadog.Trace.Security.IntegrationTests/AspNetBase.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public async Task TestBlockedRequestAsync(MockTracerAgent agent, bool enableSecu
             Func<Task<(HttpStatusCode StatusCode, string ResponseText)>> attack = () => SubmitRequest("/Health/?arg=[$slice]");
             var resultRequests = await Task.WhenAll(attack(), attack(), attack(), attack(), attack());
             agent.SpanFilters.Add(s => s.Tags["http.url"].IndexOf("Health", StringComparison.InvariantCultureIgnoreCase) > 0);
-            var spans = agent.WaitForSpans(5);
+            var spans = agent.WaitForSpans(expectedSpans);
```
Comment (robertpi) [inline test/Datadog.Trace.Security.IntegrationTests/AspNetBase.cs:69]: This is actually unrelated to the review, it should help fix the security flaky test, thanks to @kevingosse for the tip!

File: src/Datadog.Trace.ClrProfiler.Native/packages.config [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="nlohmann.json" version="3.5.0" targetFramework="native" />
-  <package id="libddwaf" version="1.0.6" targetFramework="net462" />
+  <package id="libddwaf" version="1.0.7" targetFramework="net462" />
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/packages.config:4]: Shouldn't this be `targetFramework="native"`?

File: test/test-applications/security/Samples.AspNetCore5/Properties/launchSettings.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
         "CORECLR_PROFILER_PATH": "$(ProjectDir)$(OutputPath)profiler-lib\\Datadog.Trace.ClrProfiler.Native.dll",
         "DD_VERSION": "1.0.0",
         "DD_TRACE_HEADER_TAGS": "sample.correlation.identifier, Server",
-        "DD_TRACE_DEBUG": "0"
+        "DD_TRACE_DEBUG": "1"
```
Comment (lucaspimentel) [inline test/test-applications/security/Samples.AspNetCore5/Properties/launchSettings.json:43]: Was this intentional or were you just testing locally?

**Review by lucaspimentel — APPROVED** [review]
Only a couple of non-blocking comments.

File: test/test-applications/security/Samples.AspNetCore5/Properties/launchSettings.json (hunk 3 above)
Comment (robertpi) [reply test/test-applications/security/Samples.AspNetCore5/Properties/launchSettings.json]: Good catch, local debug, fixed in 702cc956

File: src/Datadog.Trace.ClrProfiler.Native/packages.config (hunk 2 above)
Comment (robertpi) [reply src/Datadog.Trace.ClrProfiler.Native/packages.config]: Yes, fixed in 702cc956

File: src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs [hunk 4]
```diff
@@ -59,188 +58,189 @@ static Native()
...
-            PWL_AFTER_LAST,
+            DDWAF_TRACE,
+            DDWAF_DEBUG,
+            DDWAF_INFO,
+            DDWAF_WARN,
+            DDWAF_ERROR,
+            DDWAF_AFTER_LAST,
         }
 
 #pragma warning disable SA1300 // Element should begin with upper-case letter
```
Comment (zacharycmontoya) [inline src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs:142]: nit: Do you still need this?

File: src/Datadog.Trace/AppSec/Waf/Obj.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        private void Initailize()
+        {
+            if (innerObjInitalized)
+            {
+                return;
+            }
+
+            innerObj = (DdwafObjectStruct)Marshal.PtrToStructure(ptr, typeof(DdwafObjectStruct));
```
Comment (zacharycmontoya) [inline src/Datadog.Trace/AppSec/Waf/Obj.cs:67]: Do you need to set `innerObjInitialized = true` at the end of this method?

File: src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs [hunk 6]
```diff
@@ -59,188 +58,189 @@ static Native()
...
 
-        internal static void pw_clearAdditive(IntPtr context)
+        internal static void Destroy(IntPtr handle)
         {
-            clearAdditive(context);
+            DestroyField(handle);
         }
 
-        internal static PWArgs pw_getInvalid()
+        internal static void ContextDistroy(IntPtr handle)
```
Comment (zacharycmontoya) [inline src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs:179]: nit: Small typo, should be `ContextDestroy`

**Review by zacharycmontoya — APPROVED** [review]
LGTM with some small comments

File: src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs (hunk 6 above)
Comment (robertpi) [reply src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs]: fixed in bef5d961

File: src/Datadog.Trace/AppSec/Waf/Obj.cs (hunk 5 above)
Comment (robertpi) [reply src/Datadog.Trace/AppSec/Waf/Obj.cs:68]: good catch! fixed in bef5d961

File: src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs (hunk 4 above)
Comment (robertpi) [reply src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs]: you're right, removed in bef5d961

**Pull request merged**